lib/src/expression/pipe.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/syntactic_entity.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/src/dart/ast/ast.dart';
+
+/// An implementation of [Expression] for the pipe operator in AngularDart.
+class PipeExpression extends ExpressionImpl implements BinaryExpression {
```

Comment:
This makes it easier, in that I can take an existing BInaryExpression and emit something that matches that interface. Unless you're planning on evaluating the expression statically I think keeping this for now is safe.